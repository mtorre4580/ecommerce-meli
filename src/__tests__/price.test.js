import React from 'react';
import Price from '../components/commons/price/price';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('<Price />', () => {

    let value = {
        currency: 'BR',
        amount: 400
    };

    it('should render snapshot', () => {
        const component = renderer.create(<Price value={value} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be equal value props', () => {
        const wrapper = mount(<Price value={value} />);
        expect(wrapper.props().value).toEqual(value);
    });

    it('should be render currency ARS $', () => {
        value = {
            ...value,
            currency: 'ARS'
        };
        const wrapper = shallow(<Price value={value} />);
        expect(wrapper.find('.price').text()).toEqual('$400');
    });

    it('should be render currency PY U$S', () => {
        value = {
            ...value,
            currency: 'PY'
        };
        const wrapper = shallow(<Price value={value} />);
        expect(wrapper.find('.price').text()).toEqual('U$S400');
    });

    it('should be render currency PY U$S', () => {
        value = {
            ...value,
            currency: 'BR'
        };
        const wrapper = shallow(<Price value={value} />);
        expect(wrapper.find('.price').text()).toEqual('R$400');
    });

    it('should be render currency default', () => {
        delete value.currency;
        const wrapper = shallow(<Price value={value} />);
        expect(wrapper.find('.price').text()).toEqual('$400');
    });
     
 });


