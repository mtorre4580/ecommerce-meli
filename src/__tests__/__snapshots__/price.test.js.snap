// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price /> should render snapshot 1`] = `
<p
  className="price"
>
  <span>
    R$
  </span>
  400
</p>
`;
